### PR TITLE
Reverted cahced_property to Python 3.7 friendly version.

### PR DESCRIPTION
**Pull request by michalivne**
Signed-off-by: Micha Livne <mlivne@cs.toronto.edu>

# What does this PR do ?

Revert to colab Python 3.7friendly version of cached_property

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py [hunk 1]
```diff
@@ -422,7 +422,9 @@ def loss_func(output_tensor):
 
         return fwd_output_and_loss_func
 
-    @functools.cached_property
+    # property
```
Comment (Zhilin123) [inline nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py:425]: Can you try `
@functools.property
@functools.lru_cache(maxsize=None)
`?

Comment (Zhilin123) [reply nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py:425]: Each of them should be in a line of its own

**Review by Zhilin123 — CHANGES_REQUESTED** [review]
Can you try the suggestion below? Also I can't seem to fetch this branch 'michalivne:megatron-python-3.7-compatibility' on colab for testing on my side. If you can verify that this code below doesn't break your functionality, I can create an identical PR that has a branch on the NVIDIA account (instead of michalivne) so that I can test it on the colab side.

**Comment by michalivne** [comment]
A new PR has been created by @Zhilin123

**Pull request closed without merging**